### ISSUE TITLE
Wekan on Uberspace 7?

### ISSUE THREAD

**Issue by jeyca** [wekan/wekan#2009]
I was happy to see, that there is even a script for installing Wekan on an Uberspace, but since Iam using Uberspace 7, I wanted to ask if there is a way to install it there as well? As far as I know, there is no mongodb yet on U7, is it possible to use Wekan with other databases?

**Comment by xet7** [comment]
@jeyca 

No, Wekan requires MongoDB.

If Uberspace 7 can run docker containers, then you could use docker-compose.yml that is at https://github.com/wekan/wekan

There is related #787 and mention about nice-to-have other databases at [blog post](https://blog.wekan.team/2018/11/year-2018-what-we-archieved/index.html) but currently it looks like adding other database support would require a lot of work, so probably will not happen soon.

**Comment by mpc3k** [comment]
@xet7 and @jeyca 
You might want to have a look at the guide provided at [UberLab/Uberspace 7 Manual: Wekan](https://lab.uberspace.de/guide_wekan.html), which made me successfully install Wekan on a Uberspace 7 domain. A link to that in your documentation might help others, too.

I've grown to be a fond user in a very short time, but recently failed miserably at my first update procedure (to—but not specifically—v5.17), as above stated guide on UberLab only provides a tiny paragraph on a vaguely written update procedure. Maybe not vague at all to those, who are more savvy doing these kind of things. 

A detailed step-by-step paragraph on how to do update in this specific environment, would encourage others to update frequently, or install Wekan on Uberspace 7 at all. I tried to get in touch with the guide author Andrej Schoeke, but could not find any publicly visible contact address.

**Comment by xet7** [comment]
@mpc3k 

Thanks! I added it to [wiki](https://github.com/wekan/wekan/wiki/Install-latest-Wekan-release-on-Uberspace). At Wekan website https://wekan.github.io is Download section / Uberspace button that go  [Uberspace wiki page](https://github.com/wekan/wekan/wiki/Install-latest-Wekan-release-on-Uberspace).

Do you know more detailed install steps? Can you add those steps as comment to this issue?

**Comment by xet7** [comment]
For Wekan bundle version, general install info is at https://github.com/wekan/wekan/wiki/Raspberry-Pi

**Comment by mpc3k** [comment]
> Do you know more detailed install steps? 

Unfortunately, no. The steps provided at UberLab were sufficient to get _me_ started, without being proficient _at all_ with the underlying assets, nor coding. That's why I'm looking for an equally detailed guide to update my existing installation, which failed.

> Can you add those steps as comment to this issue?

If you think that helps others here, I can do that. Wouldn't that be redundant though, as we have the instructions at UberLab?

**Comment by BabyFnord** [comment]
@xet7 FYI, in cooperation with Uberspace, we now have a Bash script which eases the update procedure, for Uberspace (7) hosted domains running Wekan:
https://github.com/BabyFnord/uberspace-update_wekan

Tested working with updating from v5.17 to v5.27 … looking forward to perform coming updates :). We'd be glad if you're inclined to include this in the manual.

**Comment by tobiasgraeber** [comment]
> Edit: Tried to find out the location of the file to edit, to let Wekan send Mails, but couldn't find it. Any pointers, please?

**Works like this, added to wekan.ini in /home/YOUR-USERNAME/etc/services.d on uberspace server, added (seperated by a comma) to the "environment" string:**

`
MAIL_URL='smtp://YOUR-USERNAME@uber.space:YOUR-SSH-PASSWORD@YOUR-SERVERNAME.uberspace.de:587',MAIL_FROM='Board Support <YOUR-USERNAME@uber.space>'`

**Comment by BabyFnord** [comment]
> **Works like this, added to wekan.ini in /home/YOUR-USERNAME/etc/services.d on uberspace server, added (seperated by a comma) to the "environment" string:**

Thanks, @tobiasgraeber ! Following your directions and restartin wekan via `supervisorctl restart wekan`, my log still reads `(Mail not sent; to enable sending, set the MAIL_URL environment variable.)` :(

**Comment by tobiasgraeber** [comment]
Seems as if the setting/variable then has not been recognized yet. Can you post your (anonymized) config file? Did you run 'supervisorctl update' and 'supervisorctl reload'?

**Comment by BabyFnord** [comment]
> Did you run 'supervisorctl update' and 'supervisorctl reload'?

No, but I did run `supervisorctl restart wekan`. Ran 'supervisorctl update' and 'supervisorctl reload' just now, didn't change my issue.

This is my wekan.ini (port # is different):
[program:wekan]
environment=PORT="0000",MONGO_URL="mongodb://wekan:PASS@127.0.0.1:27017/wekan",ROOT_URL="https://USERNAME.uber.space"
MAIL_URL='smtp://USERNAME@uber.space:SSHPASSWORD@SERVER.uberspace.de:587',MAIL_FROM='Board Support <USERNAME@uber.space>'
directory=%(ENV_HOME)s/wekan/current
command=node main.js
autostart=yes
autorestart=yes

**Comment by tobiasgraeber** [comment]
Try adding a "," behind the ROOT_URLs Path and write all the environment Params in one line, like this:

```

[program:wekan]
environment=PORT="0000",MONGO_URL="mongodb://wekan:PASS@127.0.0.1:27017/wekan",ROOT_URL="https://USERNAME.uber.space"**,**MAIL_URL='smtp://USERNAME@uber.space:SSHPASSWORD@SERVER.uberspace.de:587',MAIL_FROM='Board Support USERNAME@uber.space'
directory=%(ENV_HOME)s/wekan/current
command=node main.js
autostart=yes
autorestart=yes
```

Then it should work (my stucture is, besides of that, identical and working like this).

Good luck ;-)

**Comment by BabyFnord** [comment]
That *did* change something, but we're not there yet :/ … With this alteration to wekan.ini, Wekan itself now reports
`Fehler beim Senden der E-Mail: connect EHOSTUNREACH 185.26.156.78:587`
and this hasn't happened before. Ugh …

**Comment by tobiasgraeber** [comment]
Should still be correct. The error message EHOSTUNREACH only says that the specified host could not be reached. I think your used host address or the port is not correct. Config structure should be fine. If you use the default Ports and the uberspace SMTP with the structure as i suggested i my example for the environment string, it should definitely work:

`environment=PORT="8080",MONGO_URL="mongodb://wekan:PASS@127.0.0.1:27017/wekan",ROOT_URL="https://USERNAME.uber.space",MAIL_URL='smtp://USERNAME@uber.space:SSHPASSWORD@SERVERNAME.uberspace.de:587',MAIL_FROM='Board Support USERNAME@uber.space'`

**Comment by BabyFnord** [comment]
@tobiasgraeber A non-specified server (at least not by me, as far as I recall whatever I already tried in this scope) is being called (the one used by Uberspace dashboard), that's the background of EHOSTUNREACH according to the support team. I'm trying to (hopefully) resolve the issue with their help and will report back.

**Comment by BabyFnord** [comment]
No solution available, I'm giving up here.

**Comment by tobiasgraeber** [comment]
Why you give up? It can only be a little difference in your case.

Double check these parts:

`,MAIL_URL='smtp://USERNAME@uber.space:SSH-PASSWORD@SERVERNAME:587',MAIL_FROM='Board Support <USERNAME@uber.space>'`

SSH-PASSWORD = Password set on https://dashboard.uberspace.de/dashboard/authentication (right handside)
USERNAME = Username from top of https://dashboard.uberspace.de/dashboard/datasheet
SERVERNAME = Servername from bottom of https://dashboard.uberspace.de/dashboard/datasheet (mine is "jarnsaxa.uberspace.de")

Make als sure your password/username does not contain any non-url-conform letters/signs

**Comment by BabyFnord** [comment]
The SSH PW is the one I regularly use along with the user- and servername for SSH, frequently :) All of those have been confirmed by Uberspace support, as well as my full wekan.ini details, according to your template. Uberspace support told me, Wekan tries to connect to a different Uberspace server than the one I specified → oterma.uberspace.de (which is used by Dashboard services, this one has no SMTP-Relay), when it should be wolf.uberspace.de … go figure.

Within the scope of wekan.ini, there is nothing either them (support) or me can set up differently. And the reasons for that behaviour is completely beyond me :/

**Comment by tobiasgraeber** [comment]
Ok. Sorry to hear that, then i also don't know :confused:

**Comment by BabyFnord** [comment]
@tobiasgraeber I solved my issue by chance: 
Creating a dedicated Mailbox and employing it as`new@domain.tld` with an [a-z-0-9]-only PW, finally did it. It did not work with the userspace-matching Mailbox and its generic PW 💁🏻‍♂️.

Thanks for helping me solve this, I hope it'll be useful to others too 😊.
